### ISSUE TITLE
@Persistent should not be considered when scanning for Couchbase entities

### ISSUE THREAD

**Issue by scottfrederick** [spring-projects/spring-boot#26074]
In https://github.com/spring-projects/spring-data-couchbase/issues/1121, Spring Data Couchbase will stop looking for  `@Persistent` when scanning for entities. Spring Boot's `CouchbaseDataConfiguration` should align with this change.

**Comment by scottfrederick** [comment]
Unblocked by https://github.com/spring-projects/spring-data-couchbase/commit/351370ba4d0f166a6c59c5a92bcd179bea97c688

**Comment by snicoll** [comment]
Closing in favor of PR #28124
